pyramda/iterable/flatten.py

```diff
@@ -1,37 +1,30 @@
 from collections import Iterable
 from past.builtins import basestring
 
+from pyramda.isinstance import isinstance
 from pyramda.function.curry import curry
 
 
 @curry
-def flatten(xs, depth):
+def flatten_until(is_leaf, xs):
```

Comment:
I think `flatten_until` should avoid special casing on the type of inputs and only specify its behavior in terms of Iterable. The fact that a dictionary is treated as an Iterable of its keys is pretty weird (more sensibly it should be an iterable of keys and values) but I don't think it's weirdness that `flatten_until` is responsible for handling well. There could be a separate `flatten`-like function for nested dictionary-like types.